src/notes.js

```diff
@@ -0,0 +1,128 @@
+// Importing the File System Package for File Handling Tasks
+const fs = require("fs");
+const fsPromises = fs.promises;
+
+/**
+ * loadNotes is an asynchronous function which reads the JSON
+ * File, converts to JSON and returns it.
+ * @returns {object array}
+ */
+const loadNotes = async () => {
+    try {
+        const data = await fsPromises.readFile("notes.JSON");
+        const dataJSON = data.toString();
+        return JSON.parse(dataJSON);
+    } catch (err) {
+        return [];
+    }
+};
+
+/**
+ * saveNotes is an asynchronous function which saves the notes
+ * array (consisting of objects) as a notes.JSON File.
+ * @param {object array} notes
+ */
+const saveNotes = async (notes) => {
+    const dataJSON = JSON.stringify(notes);
+    await fsPromises.writeFile("notes.JSON", dataJSON);
+};
+
+/**
+ * addNotes function calls loadNotes function which reads and
+ * loads the JSON file and then returns it. Then
+ * a check is performed to find if a note with the same title
+ * already exists. If not, then a note object is created and
+ * pushed to the notes array and then the saveNotes function
+ * is called, then a boolean true value is retuned If a duplicate
+ * note is found then a false boolean value is retuned. In case
+ * there is an error, it is logged on to the console.
+ * @param {string} title
+ * @param {string} body
+ */
+const addNotes = async (title, body) => {
+    try {
+        const notes = await loadNotes();
+        const duplicateNote = notes.find((note) => note.title === title);
+        if (!duplicateNote) {
+            const timestamp = new Date().toLocaleString("en-US", {
+                timeZone: "Asia/Kolkata",
+            });
+            notes.push({
+                title: title,
+                body: body,
+                createdAt: timestamp,
+                lastModifiedAt: timestamp,
+            });
+            await saveNotes(notes);
+            return true;
+        }
+        return false;
+    } catch (err) {
+        console.error(err);
```

Comment:
Return Promise reject in catch block, so that the calling function can decide what can be done.

Reply:
Fixed this. Kindly review!